nextflow-parsing/NextflowParsingFunction/src/main/java/io/dockstore/nextflowparsing/NextflowUtilities.java

```diff
@@ -0,0 +1,225 @@
+   * this helps us https://github.com/ninjudd/drip
+   *
+   * @param content a file object with the content of nextflow
+   * @return a commons configuration file with the keys from the nextflow config file
+   */
+  public static Configuration grabConfig(File content) {
+    try {
+      final List<String> strings =
+          Arrays.asList(
+              "java", "-jar", getNextflowTargetFile().getAbsolutePath(), "config", "-properties");
+      String stdout = executeNextflowConfig(content.getParentFile(), strings);
+      Properties properties = new Properties();
+      properties.load(new StringReader(stdout));
+      return ConfigurationConverter.getConfiguration(properties);
+    } catch (RuntimeException | IOException | InterruptedException e) {
+      LOG.error("Problem running Nextflow: ", e);
+      throw new NextflowParsingException("Could not run Nextflow", e);
+    }
+  }
+
+  /**
+   * Converts the contents of a config file to a commons config file.
+   *
+   * @param content the content of the config file
+   * @return a commons configuration file with the keys from the nextflow config file
+   */
+  public static Configuration grabConfig(String content) {
+    Path nextflowDir = null;
+    try {
+      // FIXME: this sucks, but we need to ignore includeConfig lines.
+      //  We basically have a chicken and the egg problem
+      // FIXME: the nextflow config command only works when all included files are present,
+      //  however we're trying to use the nextflow config command to figure out what the list of
+      //  included files is to determine what files we want to get from the GitHub API in the first
+      //  place
+      // FIXME: secondary case: when looking for description and author,
+      //  we don't actually need includes either
+      String newContent = content.replaceAll("(?i)(?m)^[ \t]*includeConfig.*", "");
+      // needed since Nextflow binary assumes content is in working directory
+      nextflowDir = Files.createTempDirectory("nextflow");
+      final Path tempFile = Paths.get(nextflowDir.toString(), "nextflow.config");
+      Files.write(tempFile, newContent.getBytes(StandardCharsets.UTF_8));
+      return grabConfig(tempFile.toFile());
+    } catch (IOException e) {
+      throw new NextflowParsingException("unable to parse nexflow config");
+    } finally {
+      if (nextflowDir != null) {
+        FileUtils.deleteQuietly(nextflowDir.toFile());
+      }
+    }
+  }
+
+  /**
+   * This is an expensive operation; a new Java VM is spun up for this, so only allow one at a time
+   * by synchronizing the method.
+   *
+   * @param directory the directory to execute the command
+   * @param commands the command
+   * @return nothing
+   */
+  private static synchronized String executeNextflowConfig(File directory, List<String> commands)
+      throws InterruptedException, IOException {
+    ProcessBuilder builder = new ProcessBuilder();
```

Comment:
Wouldn't it be better to run as a library, rather than bringing up a new VM?

I forget, why did we do it this way for the web service? Because it was chewing up too much of the VM's heap? If it's a spin-up/spin-down lambda, it shouldn't matter.

Reply:
Correction to the correction, it looks like we always use the `DEFAULT_NEXTFLOW_VERSION` -- `getNextflowTargetFile(INIConfiguration config)` is never called, neither here nor in dockstore/dockstore, so I go back to thinking this should be called as a method.

Reply:
Not actually sure why. Seems fairly simple upon trying it out right now.

Reply:
We might have done it this way for the web service to avoid conflicts between our classpath and Nextflow's classpath. That shouldn't be a concern here. Or at least not as much